contracts/mainnet/connectors/euler/main.sol

```diff
@@ -95,12 +95,14 @@ abstract contract Euler is Helpers {
 
 		TokenInterface tokenContract = TokenInterface(_token);
 		IEulerEToken eToken = IEulerEToken(markets.underlyingToEToken(_token));
-		_amt = _amt == uint256(-1) ?  eToken.balanceOf(address(this)) : _amt;
-		uint256 initialBal = tokenContract.balanceOf(address(this));
+
+		address _subAccount = getSubAccount(address(this), subAccount);
+		_amt = _amt == uint256(-1) ?  eToken.balanceOfUnderlying(_subAccount) : _amt;
+		uint256 initialBal = tokenContract.balanceOf(_subAccount);
```

Comment:
@shriyatyagii This should be `address(this)` only

Reply:
We are checking withdrawal amount to dsa. not to subaccount address

Reply:
Ah! okay